test/int/adapter.test.ts

```diff
@@ -72,18 +85,49 @@ suite('Chrome Debug Adapter etc', function () {
         });
 
         test('should stop on debugger statement in http://localhost', () => {
-            const testProjectRoot = path.join(DATA_ROOT, 'intervalDebugger');
             const breakFile = path.join(testProjectRoot, 'src/app.ts');
             const DEBUGGER_LINE = 2;
 
-            server = createServer({ root: testProjectRoot });
-            server.listen(7890);
-
             return Promise.all([
                 dc.configurationSequence(),
                 dc.launch({ url: 'http://localhost:7890', webRoot: testProjectRoot }),
                 dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
             ]);
         });
+
+        test('Should attach to existing instance of chrome and break on debugger statement', async () => {
+            const breakFile = path.join(testProjectRoot, 'src/app.ts');
+            const DEBUGGER_LINE = 2;
+
+            const browser = await puppeteer.launch({ headless: false, args: ['http://localhost:7890', '--remote-debugging-port=7777'] });
+
+            await Promise.all([
+                dc.configurationSequence(),
+                dc.initializeRequest().then(_ => {
+                    return dc.attachRequest(<IAttachRequestArgs>{ url: 'http://localhost:7890', port: 7777, webRoot: testProjectRoot });
+                }),
+                dc.assertStoppedLocation('debugger_statement', { path: breakFile, line: DEBUGGER_LINE } )
+            ]);
+
+            await browser.close();
```

Comment:
It might be a good idea to put the .close() in a finally block, so we'll close the browser even if the test fails...

Reply:
Good suggestion. I added a finally block for both tests that close with puppeteer.